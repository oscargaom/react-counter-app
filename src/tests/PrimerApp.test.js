import React from 'react'
import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import { render } from "@testing-library/react";
import { shallow } from 'enzyme';

describe("Pruebas en el componente <PrimerApp />", () => {
//   test("Debe encontrar el texto: 'Hola, Soy Goku'", () => {
    
//     const saludo = 'Hola, Soy Goku';
//     const { getByText } = render(<PrimeraApp saludo={saludo} />);

//     expect(getByText(`${saludo}`)).toBeInTheDocument();
    
//   });

    test("Debe encontrar el texto: 'Hola, Soy Goku'", () => {
        const saludo = "Hola, Soy Goku";

        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();

    });

    test("Debe validar que el subtitulo sea el indicado", () => {
        const saludo = "Hola, Soy Goku";
        const subtitulo = 'Serie: Dragon Ball Z';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo}
            />);

        const texto = wrapper.find('p').text();

        expect(texto).toBe(subtitulo);
        
    });

});
