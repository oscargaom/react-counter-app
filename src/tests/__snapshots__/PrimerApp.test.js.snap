// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <PrimerApp /> Debe encontrar el texto: 'Hola, Soy Goku' 1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
